R/likelihood.R

```diff
@@ -78,9 +132,38 @@ likelihood <- function(chains, statistic = c("size", "length"), offspring_dist,
   checkmate::assert_numeric(
     exclude, null.ok = TRUE
   )
+  # likelihood cannot work with an <epichains> object so convert to
+  # <epichains_summary>
+  if (.is_epichains(chains)) {
+    chains <- summary(chains)
+  }
+
+  # Logic:
+  # 1. If stat_threshold is not specified:
+  #
+  # * If the object is an epichains_summary, we'll give preference to the
+  # stat_threshold used in the simulation and stored as an attribute in the
+  # object.
+  #
+  # * If the object is numeric, we'll use the default of Inf.
+  #
+  # 2. If stat_threshold is specified, we'll give that precedence.
+  if (missing(stat_threshold) && .is_epichains_summary(chains)) {
+      stat_threshold <- attr(chains, "stat_threshold")
+  }
+ # chains must be at most stat_threshold
+  if (is.finite(stat_threshold)) {
+    chains <- pmin(chains, stat_threshold)
+  }
```

Comment:
Yes, I fixed it previously. See here e7889a1849779bd49915c47f4e6b978a4c203774.

I had to check if the object is numeric but checking that it's not `<epichains_summary>` because `<epichains_summary>` objects are also numeric and will pass the `is.numeric()` check and hence, error here.